### PR TITLE
Makefile: fix cross-compiling issues

### PR DESCRIPTION
**Pull request by liuming50**
To be able to support cross compilation, some fixes are needed as
follows:
- Replace all the host paths with a prefix sysroot, it defaults to be
  empty.
- Use weak assignment to compiler variables, like CC, this allows them
  to be overridden by environment variables passed to make.

Signed-off-by: Ming Liu <liu.ming50@gmail.com>

### CODE REVIEW

**Comment by NanlinXie** [comment]
Can one of the admins verify this patch?

File: Makefile [hunk 1]
```diff
@@ -12,6 +12,8 @@ TOOLS_OUT := $(ROOT_OUT)/tools
 DOC_OUT := $(ROOT_OUT)/doc
 BUILD_VERSION ?=
 BUILD_TAG ?=
+DESTDIR ?=
```
Comment (binbinwu1) [inline Makefile]: This var not used?

Comment (liuming50) [reply Makefile]: It's being referred by install targets in Makefiles, such as:
...
./devicemodel/Makefile:	install -D --mode=0755 $(DM_OBJDIR)/$(PROGRAM) $(DESTDIR)/usr/bin/$(PROGRAM)
./devicemodel/Makefile:	install -D -t $(DESTDIR)/usr/share/acrn/samples/nuc $^
./devicemodel/Makefile:	install -D -t $(DESTDIR)/usr/share/acrn/samples/apl-mrb $^
./devicemodel/Makefile:	install -d $(DESTDIR)/usr/lib/systemd/system/
./devicemodel/Makefile:	install -p -D -m 0644 ./samples/apl-mrb/acrn_guest.service $(DESTDIR)/usr/lib/systemd/system
./devicemodel/Makefile:	install -d $(DESTDIR)/usr/share/acrn/bios
./devicemodel/Makefile:	install -D --mode=0664 -t $(DESTDIR)/usr/share/acrn/bios $^
...

here I just want give it a default empty value in the top Makefile, it's not a bug fix, but I think it would be easier to be read.

**Comment by binbinwu1** [comment]
Reviewed-by: Binbin Wu <binbin.wu@intel.com>

**Comment by liuming50** [comment]
Will send a V2 with "Tracked-On" tag.

File: Makefile (hunk 1 above)
Comment (gvancuts) [reply Makefile]: `DESTDIR` should not be directly set in the `Makefile`, it is a built-in variable that is only used (and only useful) to the `(un)install` targets of a `Makefile` - see https://www.gnu.org/prep/standards/html_node/DESTDIR.html#DESTDIR

You're not setting it to any value but it's awkward to have it in the Makefile, it is expected that readers of Makefiles know this is a built-in variable. Can you remove it in your v2?

**Comment by gvancuts** [comment]
> Will send a V2 with "Tracked-On" tag.

You should force-push an updated version of your patch, else it becomes really hard to follow the history of what happened to a particular conversation. I don't know for example if you have already sent a v2 patch. Can you make sure that you update (and force-push) your patches to as to re-use the same PR until the code gets merged? Thanks!

(if you need help/instructions on how to do that, we have some here: https://projectacrn.github.io/latest/developer-guides/contribute_guidelines.html#contribution-workflow)

File: Makefile (hunk 1 above)
Comment (liuming50) [reply Makefile]: You are absolutely right, will remove it.

**Comment by liuming50** [comment]
OK, will try to do that. thanks for the explanation. But I had sent out a V2, will send a V3 on top of that one.
